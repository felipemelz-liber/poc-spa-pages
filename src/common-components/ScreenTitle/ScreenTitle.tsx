import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Icon from '@mdi/react';
import { mdiArrowLeft } from '@mdi/js';
import { handleThemeFromObject as getTheme } from 'liber-components';

const Container = styled.span`
  font-family: Roboto;
  font-size: 32px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.41;
  letter-spacing: 0.13px;
  color: ${({ theme }) => getTheme(theme, 'colors.primary', '#009dff')};
  margin: 24px 0px;

  width: 100%;
  display: flex;
  align-items: center;

  & > svg {
    width: 32px;
    height: 32px;
    fill: ${({ theme }) => getTheme(theme, 'colors.primary', '#009dff')};
    margin-right: 16px;
    margin-bottom: 2px;
    cursor: pointer;
  }
`;

const ScreenTitle = ({ handleBack, children }) => (
  <Container onClick={handleBack}>
    {handleBack && <Icon data-testid="page-back-button"  path={mdiArrowLeft} />}
    {children}
  </Container>
);

ScreenTitle.propTypes = {
  handleBack: PropTypes.func,
  children: PropTypes.node,
};

ScreenTitle.defaultProps = {
  handleBack: null,
  children: <div />,
};

export default ScreenTitle;
