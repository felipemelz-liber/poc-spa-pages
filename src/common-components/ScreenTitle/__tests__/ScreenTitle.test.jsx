import React from 'react';
import sinon from 'sinon';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Icon from '@mdi/react';
import ScreenTitle from '../ScreenTitle';

describe('ScreenTitle component tests', () => {
  configure({ adapter: new Adapter() });

  const handleBack = sinon.spy();

  it('should render ScreenTitle correctly without props', () => {
    const wrapper = shallow(<ScreenTitle />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render ScreenTitle correctly with props', () => {
    const wrapper = shallow(<ScreenTitle handleBack={handleBack}>Mock Title</ScreenTitle>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onFilter when clicking on FilterButton', () => {
    const wrapper = shallow(<ScreenTitle handleBack={handleBack}>Mock Title</ScreenTitle>);
    expect(toJson(wrapper)).toMatchSnapshot();
    const backButton = wrapper.find(Icon);
    backButton.simulate('click');
    expect(handleBack.called).toBeTruthy();
  });
});
