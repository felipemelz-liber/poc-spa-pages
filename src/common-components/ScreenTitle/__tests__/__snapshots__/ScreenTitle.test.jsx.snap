// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenTitle component tests should call onFilter when clicking on FilterButton 1`] = `
<styled.span>
  <Icon
    color="currentColor"
    data-testid="page-back-button"
    horizontal={false}
    onClick={[Function]}
    path="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
    rotate={0}
    size={null}
    spin={false}
    vertical={false}
  />
  Mock Title
</styled.span>
`;

exports[`ScreenTitle component tests should render ScreenTitle correctly with props 1`] = `
<styled.span>
  <Icon
    color="currentColor"
    data-testid="page-back-button"
    horizontal={false}
    onClick={[Function]}
    path="M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z"
    rotate={0}
    size={null}
    spin={false}
    vertical={false}
  />
  Mock Title
</styled.span>
`;

exports[`ScreenTitle component tests should render ScreenTitle correctly without props 1`] = `
<styled.span>
  <div />
</styled.span>
`;
